analyses/org.eclipse.tracecompass.incubator.executioncomparision.ui/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/ui/DifferentialFlameGraphView.java

```diff
@@ -0,0 +1,1261 @@
+                    break;
+                }
+            }
+        }
+    }
+
+    private static @Nullable ITmfTrace getCurrentTrace(@Nullable IEditorPart editor) {
+        ITmfTrace trace = null;
+        if (editor instanceof ITmfTraceEditor tmfTraceEditor) {
+            trace = tmfTraceEditor.getTrace();
+        } else {
+            // Get the active trace, the editor might be opened on a script
+            trace = TmfTraceManager.getInstance().getActiveTrace();
+        }
+        return trace;
+    }
+
+    /**
+     * Get the time graph viewer
+     *
+     * @return the time graph viewer
+     */
+    public TimeGraphViewer getTimeGraphViewer() {
+        Objects.requireNonNull(fTimeGraphViewer);
+        return fTimeGraphViewer;
+    }
+
+    /**
+     * Handler for the trace selected signal
+     *
+     * @param signal
+     *            The incoming signal
+     */
+    @TmfSignalHandler
+    public void traceSelected(final TmfTraceSelectedSignal signal) {
+        try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINE, "DifferentialFlameGraphView::traceSelected")) { //$NON-NLS-1$
+            ITmfTrace trace = signal.getTrace();
+
+            fTrace = trace;
+            if (trace == null) {
+                return;
+            }
+            // If entries for this trace are already available, just zoom on
+            // them,
+            // otherwise, rebuild
+            List<TimeGraphEntry> list = fEntryListMap.get(trace);
+            if (list == null) {
+                refresh();
+                Display.getDefault().asyncExec(() -> buildFlameGraph(trace, null, null));
+
+            } else {
+                // Reset end time
+                long endTime = Long.MIN_VALUE;
+                for (TimeGraphEntry entry : list) {
+                    endTime = Math.max(endTime, entry.getEndTime());
+                }
+                setEndTime(endTime);
+                refresh();
+                startZoomThread(0, endTime, false);
+            }
+        }
+    }
+
+    /**
+     * Get the callgraph modules used to build the view
+     *
+     * @return The call graph provider modules
+     */
+    protected Iterable<ICallGraphProvider2> getCallgraphModules() {
+        ITmfTrace trace = fTrace;
+        if (trace == null) {
+            return Collections.emptyList();
+        }
+        String analysisId = NonNullUtils.nullToEmptyString(getViewSite().getSecondaryId());
+        Iterable<ICallGraphProvider2> modules = TmfTraceUtils.getAnalysisModulesOfClass(trace, ICallGraphProvider2.class);
+        return StreamSupport.stream(modules.spliterator(), false)
+                .filter(m -> {
+                    if (m instanceof IAnalysisModule analysisModule) {
+                        return analysisModule.getId().equals(analysisId);
+                    }
+                    return true;
+                })
+                .collect(Collectors.toSet());
+    }
+
+    private class BuildRunnable {
```

Comment:
Refactor this private class in another file to reduce the complexity of the DifferentialFlameGraphView.